include/tvm/relay/expr_functor.h

```diff
@@ -232,6 +232,181 @@ class ExprMutator
   std::unordered_map<Expr, Expr, ObjectHash, ObjectEqual> memo_;
 };
 
+/*!
+ * \brief A wrapper around ExprVisitor which traverses the Dataflow Normal AST.
+ *
+ * DataflowVisitor treats Expr as dataflow graph, and visits in post-DFS order
+ *
+ * DataflowVisitor provides the same recursive API as ExprVisitor, and uses
+ * recursion to traverse most forms of the IR, but under the hood it expands nested dataflow regions
+ * of the graph and processes them iteratatively to prevent stack overflows
+ */
+class DataflowVisitor : public ::tvm::relay::ExprVisitor {
```

Comment:
Shall we unify the naming convention of DataflowVisitor and ScopeMutator? Perhaps ScopeVisitor?

Reply:
I don't think ScopeVisitor is a very good name, i.e what is the property that the visitor is maintaining respecting? it is not just scoping.

Reply:
MixedModeVisitor?